src/utils/report.js

```diff
@@ -57,6 +57,7 @@ export default function ({
 
   const severity = get(result.stylelint, [ "ruleSeverities", ruleName ], "ignore")
 
+  /* istanbul ignore if */
   if (typeof severity === "undefined") {
```

Comment:
Yes. It's not currently tested, and as you pointed out can't actually happen. So let's cut it.